src/Utilities/CssConcatenator/RuleBlock.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pelago\Emogrifier\Utilities\CssConcatenator;
+
+/**
+ * This class represents a CSS rule block, i.e., one ore multiple selectors (like "p") and a declarations block
+ * (like "margin-top: 2pc").
+ *
+ * @internal
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ * @author SignpostMarv
+ * @author Oliver Klee <github@oliverklee.de>
+ */
+class RuleBlock
+{
+    /**
+     * selectors for the rule block
+     *
+     * @var string[]
+     */
+    private $selectors = [];
```

Comment:
I'm wondering whether trying to also maintain this property would cause efficiency issues, e.g. updating it after every merge from an increasingly growing list of selectors that have come from something like a CSS reset (having been split up in original parsing).  Would it be better to 'lazily' obtain this (via `array_keys`) when it's finally needed, i.e. when outputting the final CSS?